provers/cairo-prover-cli/src/main.rs

```diff
@@ -5,42 +5,107 @@ use lambdaworks_math::field::fields::fft_friendly::stark_252_prime_field::Stark2
-    ProveAndVerify(ProveAndVerifyArgs),
+/// Attemps to compile the Cairo program with `cairo-compile`
+/// and then save it to the desired path.  
+/// Returns `Ok` on success else returns `Error`
+fn cairo_compile(program_path: &String, out_file_path: &String) -> Result<(), Error> {
+    let out_file = File::create(out_file_path)?;
+
+    match Command::new("cairo-compile")
+        .arg("--proof_mode")
+        .arg(program_path)
+        .stderr(Stdio::null())
+        .stdout(out_file)
+        .spawn()
+    {
+        Ok(mut child) => {
+            // wait for spawned proccess to finish
+            match child.wait() {
+                Ok(_) => Ok(()),
+                Err(err) => Err(err),
+            }
+        }
+        Err(err) => Err(err),
+    }
 }
 
-#[derive(Args, Debug)]
-struct ProveArgs {
-    program_path: String,
-    proof_path: String,
+/// Attemps to compile the Cairo program with `docker`
+/// and then save it to the desired path.  
+/// Returns `Ok` on success else returns `Error`
+fn docker_compile(program_path: &String, out_file_path: &String) -> Result<(), Error> {
+    let out_file = File::create(out_file_path)?;
+    let root_dir = get_root_dir()?;
+    match Command::new("docker")
+        .arg("run")
+        .arg("--rm")
+        .arg("-v")
+        .arg(format!("{}/:/pwd", root_dir))
+        .arg("cairo")
+        .arg("--proof_mode")
+        .arg(format!("/pwd/{}", program_path))
+        .stderr(Stdio::null())
+        .stdout(out_file)
+        .spawn()
+    {
+        Ok(mut child) => {
+            // wait for spawned proccess to finish
+            match child.wait() {
+                Ok(status) => match status.code() {
+                    Some(0) => Ok(()), // exit success
+                    _ => Err(Error::new(
+                        ErrorKind::Other,
+                        "File provided is not a Cairo uncompiled",
+                    )),
+                },
+                Err(err) => Err(err),
+            }
+        }
+        Err(err) => Err(err),
+    }
 }
 
-#[derive(Args, Debug)]
-struct VerifyArgs {
-    proof_path: String,
-}
+/// Attemps to compile the Cairo program
+/// with either `cairo-compile` or `docker``
+fn try_compile(program_path: &String, out_file_path: &String) -> Result<(), Error> {
+    if !program_path.contains(".cairo") {
+        println!("The program provided is not an uncompiled .cairo program");
+        return Err(Error::new(
+            ErrorKind::Other,
+            "File provided is not a Cairo uncompiled",
+        ));
+    }
 
-#[derive(Args, Debug)]
-struct ProveAndVerifyArgs {
-    program_path: String,
+    if cairo_compile(program_path, out_file_path).is_ok()
+        || docker_compile(program_path, out_file_path).is_ok()
+    {
+        println!("Compiled cairo program");
+        Ok(())
+    } else {
+        println!("Failed to compile cairo program\nEnsure cairo-compile or docker is installed (and running)");
```

Comment:
```suggestion
        println!("Failed to compile cairo program\nEnsure either cairo-compile is installed, or Docker is running and the compiler image is built");
```

Reply:
Removed print altogether and modified `Error` return after highlighted code:

```rs
        Err(Error::new(
            ErrorKind::Other,
            "Failed to compile cairo program, neither cairo-compile nor docker found",
        ))
```

Which is then printed out by the caller.